Add support for new flags XATTR_CREATE and XATTR_REPLACE

aravindavk opened the pull request:
* `XATTR_CREATE` - set the value, fail if attr already exists.
* `XATTR_REPLACE` - set the value, fail if attr does not exist.

Signed-off-by: Aravinda Vishwanathapura <mail@aravindavk.in>

aravindavk commented:
@ettomatic Please review. Not sure why Travis CI is not triggered for this PR.

ettomatic commented on src/xattr/platforms/darwin.cr:
```diff
@@ -11,15 +11,20 @@ module XAttr
 
-      def self.set(path, key, value, size, no_follow)
-        options = no_follow ? XATTR_NOFOLLOW : 0
+      def self.set(path, key, value, size, no_follow, only_create, only_replace)
```
Worth raising an exception if for any reason both `only_create` and `only_replace` get set to `true`?

aravindavk replied:
Updated the PR. Filesystem already raises exception if both are specified(In darwin it is `EINVAL` and in linux it is `EEXIST` if attr exists and `ENODATA` if attr not exists). Added tests to confirm this behaviour.

aravindavk commented:
Updated the PR. Filesystem already raises exception if both are specified(In darwin it is `EINVAL` and in linux it is `EEXIST` if attr exists and `ENODATA` if attr not exists). Added tests to confirm this behaviour.

Let me know if anymore changes required. Thanks.

ettomatic reviewed (approved):
Thanks for this @aravindavk looks good! 👍
